lib/engine/step/g_1828/merger.rb

```diff
@@ -0,0 +1,448 @@
+# frozen_string_literal: true
```

Comment:
Contains the merge state machine that allows the player to select the merge target, creates the merged corporation, and handles share exchanges, and complete the merger. The exchange logic may require decisions by the players (who to trade a share with, which share to keep if an odd percent owned) and is designed to halt when a player choice is required and then pick back up after it is made.

Waiting on a few rule updates from JC and the multi-turn undo to completely finish and go to pre-alpha.

Reply:
Implemented based on a conversation with JC. Will tweak later if rules differ.